pkg/controller/external_secrets/networkpolicy.go

```diff
@@ -0,0 +1,217 @@
+
+	fetched := &networkingv1.NetworkPolicy{}
+	exists, err := r.Exists(r.ctx, client.ObjectKeyFromObject(networkPolicy), fetched)
+	if err != nil {
+		return common.FromClientError(err, "failed to check existence of network policy %s", networkPolicyName)
+	}
+
+	if exists {
+		if externalSecretsConfigCreateRecon {
+			r.eventRecorder.Eventf(esc, corev1.EventTypeWarning, "ResourceAlreadyExists", "NetworkPolicy %s already exists", networkPolicyName)
+		}
+		if common.HasObjectChanged(networkPolicy, fetched) {
+			r.log.V(1).Info("NetworkPolicy modified, updating", "name", networkPolicyName)
+			if err := r.UpdateWithRetry(r.ctx, networkPolicy); err != nil {
+				return common.FromClientError(err, "failed to update network policy %s", networkPolicyName)
+			}
+			r.eventRecorder.Eventf(esc, corev1.EventTypeNormal, "Reconciled", "NetworkPolicy %s updated", networkPolicyName)
+		} else {
+			r.log.V(4).Info("NetworkPolicy already up-to-date", "name", networkPolicyName)
+		}
+	} else {
+		if err := r.Create(r.ctx, networkPolicy); err != nil {
+			return common.FromClientError(err, "failed to create network policy %s", networkPolicyName)
+		}
+		r.eventRecorder.Eventf(esc, corev1.EventTypeNormal, "Reconciled", "NetworkPolicy %s created", networkPolicyName)
+	}
+
+	return nil
+}
+
+// buildNetworkPolicyFromConfig constructs a NetworkPolicy object from the API configuration.
+func (r *Reconciler) buildNetworkPolicyFromConfig(esc *operatorv1alpha1.ExternalSecretsConfig, npConfig operatorv1alpha1.NetworkPolicy, resourceLabels map[string]string) *networkingv1.NetworkPolicy {
+	namespace := externalsecretsDefaultNamespace
+	if esc.Spec.ApplicationConfig.OperatingNamespace != "" {
+		namespace = esc.Spec.ApplicationConfig.OperatingNamespace
+	}
+
+	// Determine pod selector based on component name
+	podSelector := r.getPodSelectorForComponent(npConfig.ComponentName)
+
+	// Build the NetworkPolicy object
+	networkPolicy := &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      npConfig.Name,
+			Namespace: namespace,
+			Labels:    resourceLabels,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: podSelector,
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+			},
+			Egress: npConfig.Egress,
+		},
+	}
+
+	return networkPolicy
+}
+
+// getPodSelectorForComponent returns the appropriate pod selector for the given component.
+func (r *Reconciler) getPodSelectorForComponent(componentName operatorv1alpha1.ComponentName) metav1.LabelSelector {
+	switch componentName {
+	case operatorv1alpha1.CoreController:
+		return metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app.kubernetes.io/name": "external-secrets",
+			},
+		}
+	case operatorv1alpha1.BitwardenSDKServer:
+		return metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"app.kubernetes.io/name": "bitwarden-sdk-server",
+			},
+		}
+	default:
```

Comment:
Lets us not assume, since ComponentName is mandatory. Instead we should return an error in default.